module4/lessons/Week1/JavaScript.md

```diff
@@ -0,0 +1,387 @@
+
+JavaScript can be _client-side_ and _server-side_, meaning that it can be used to control user-facing interfaces as well as handle the server-side extensions that connect with a database.
+
+It's a highly versatile and flexible language, and has become the most commonly used language of the web.
+
+Now that we know a little bit about JavaScript, let's talk about how to actually use it!
+
+# Data Types
+
+There are different data types in JavaScript. It's important to understand what type of data you're dealing with as you're writing code, and knowing the types of data are available to you is the first step.
+
+There are three basic data types: numeric, string, and boolean. Let's go over what each of these is.
+
+## Numeric Data Type
+
+This data type handles numbers. In JavaScript, numbers are written without commas so three thousand four hundred and seventy five would be written like this:
+
+`3475`
+
+Numbers can also be negative and/or decimals (also called "floats"), so we can write `-3475` and `0.5`
+
+Note that we _do not_ enclose a number data type in quotes.
+
+## String Data Type
+
+The string data type handles letters and other characters. It is always enclosed in quotes, and looks like this:
+
+`"I am a string!"`
+
+You can use either single or double quotes to wrap a string, but the opening and closing quotes need to match. A good rule of thumb is to pick a type of quote to use and then be consistent in it's use throughout a project.
+
+Strings can be used for any kind of text. We use them regularly to add text to a page.
+
+## Boolean Data Type
+
+A boolean has one of two values: `true` or `false`. Think of it like a light switch: it's either on or off.
+
+Booleans can be very useful to help determine which part of a script should be run (as in conditionals, which we will talk about later).
+
+# ES5 vs. ES6
+
+ES6 is the most current version of the ECMA Script Spec. We will be using the ES6 syntax.
+
+# Variables in JavaScript
+
+As we're writing code, there are many times when we want to be able to store a value so we have ability to reuse it in multiple places. Writing the same thing over and over is a pain, and it's common that we need to have access to the same information more than once. A variable lets us do just that!
+
+## What is a variable?
+
+A variable is a place to store values. When we write scripts (a set of instructions for a computer to follow), we need to temporarily store small pieces of data. We store that data in variables. "Variable" is a good name for this concept because it indicates the stored data can change (or _vary_) each time a script is run.
```

Comment:
Re-reading back through this, I think this might feel a bit patronizing to students because they already are very familiar with what a variable is from C#. I'm confused why BE is using this type of language in a later mod.